File: airmeishi/Models/ScanLanguage.swift
//
//  ScanLanguage.swift
//  airmeishi
//
//  Language selection for OCR scanning
//

import Foundation

/// Supported languages for OCR scanning
enum ScanLanguage: String, CaseIterable, Identifiable {
  case traditionalChinese = "zh-Hant"
  case japanese = "ja"
  case english = "en"

  var id: String { rawValue }

  var displayName: String {
    switch self {
    case .traditionalChinese:
      return String(localized: "Traditional Chinese")
    case .japanese:
      return String(localized: "Japanese")
    case .english:
      return String(localized: "English")
    }
  }

  var flag: String {
    switch self {
    case .traditionalChinese:
      return "🇹🇼"
    case .japanese:
      return "🇯🇵"
    case .english:
      return "🇺🇸"
    }
  }

  /// Get the language code for Vision framework
  var visionLanguageCode: String {
    return rawValue
  }

  /// Get all language codes for Vision framework
  static var allVisionLanguageCodes: [String] {
    return allCases.map { $0.visionLanguageCode }
  }
}

File: apps/expo/src/components/ocr/scanLanguage.ts
/**
 * scanLanguage — port of solidarity/Models/ScanLanguage.swift.
 *
 * Languages supported by the OCR scanner. The string values mirror the
 * Apple Vision language codes so a future native bridge can pass them
 * through verbatim (`zh-Hant`, `ja`, `en`).
 */
export type ScanLanguage = 'zh-Hant' | 'ja' | 'en';

export const SCAN_LANGUAGES: readonly ScanLanguage[] = [
  'zh-Hant',
  'ja',
  'en',
] as const;

export function scanLanguageDisplayName(lang: ScanLanguage): string {
  switch (lang) {
    case 'zh-Hant':
      return 'Traditional Chinese';
    case 'ja':
      return 'Japanese';
    case 'en':
      return 'English';
  }
}

export function scanLanguageFlag(lang: ScanLanguage): string {
  switch (lang) {
    case 'zh-Hant':
      return '\u{1F1F9}\u{1F1FC}'; // TW
    case 'ja':
      return '\u{1F1EF}\u{1F1F5}'; // JP
    case 'en':
      return '\u{1F1FA}\u{1F1F8}'; // US
  }
}
